frontend/src/metabase/containers/Form.jsx

```diff
@@ -227,8 +227,12 @@ class Form extends React.Component {
         const errorNames = Object.keys(error.data.errors);
         const hasUnknownFields = errorNames.some(name => !fieldNames.has(name));
         throw {
-          _error: hasUnknownFields ? t`An error occurred` : null,
-          ...error.data.errors,
+          _error:
+            error.data?.message ||
+            error.message ||
+            error.data ||
+            (hasUnknownFields ? t`An error occurred` : null),
+          ...getFieldErrors(error.data.errors, formObject.fields(values)),
```

Comment:
Kind of a hack. But this API is required for database form. See the code comment here 
https://github.com/metabase/metabase/blob/9756b27e9ad416388e950980fec94f8ec9498fad/frontend/src/metabase/containers/Form.jsx#L368